crates/aptos-rosetta/src/block.rs

```diff
@@ -44,42 +43,78 @@ async fn block(request: BlockRequest, server_context: RosettaContext) -> ApiResu
     let rest_client = &server_context.rest_client;
 
     // Retrieve by block or by hash, both or neither is not allowed
-    let response = match (
+    let (parent_transaction, transaction): (Transaction, _) = match (
         &request.block_identifier.index,
         &request.block_identifier.hash,
     ) {
-        (Some(version), None) => rest_client.get_transaction_by_version(*version).await?,
+        (Some(version), None) => {
+            // For the genesis block, we populate parent_block_identifier with the
+            // same genesis block. Refer to
+            // https://www.rosetta-api.org/docs/common_mistakes.html#malformed-genesis-block
+            if *version == 0 {
+                let response = rest_client.get_transaction_by_version(*version).await?;
+                let txn = response.into_inner();
+                (txn.clone(), txn)
+            } else {
+                let response = rest_client
+                    .get_transactions(Some(*version - 1), Some(2))
```

Comment:
Can you document why we're fetching exactly 2 transactions here?

Reply:
I guess it is because we need txn of current version and its parent

Reply:
We need the previous "block" for every "block".  In this case a "block" is a transaction.  So, we need to request 2.

Reply:
I'm not a Rust expert but I think this function is too complex with matching responses from 2 different cases. Can we just have 2 functions, one fetching the genesis transaction, and one always fetching and processing the response for 2 transactions?